consensus/hotstuff/pacemaker/pacemaker.go

```diff
@@ -86,7 +89,10 @@ func (p *ActivePaceMaker) LastViewTC() *flow.TimeoutCertificate { return p.viewT
 func (p *ActivePaceMaker) TimeoutChannel() <-chan time.Time { return p.timeoutControl.Channel() }
 
 // BlockRateDelay returns the delay for broadcasting its own proposals.
-func (p *ActivePaceMaker) BlockRateDelay() time.Duration { return p.timeoutControl.BlockRateDelay() }
+// todo rename?
+func (p *ActivePaceMaker) BlockRateDelay() time.Duration {
```

Comment:
At this point I would lean into removing the `BlockRateDelay` completely from pacemaker and using provider in `EventHandler`. Pacemaker has nothing to do with it and just proxies call to provider, it has zero overlap with logic of pacemaker, previously we at least shared a config but not anymore.